eos/utils/omnes-factor.hh

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2024 Florian Herren
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_UTILS_OMNES_FACTOR_HH
+#define EOS_GUARD_EOS_UTILS_OMNES_FACTOR_HH 1
+
+#include <eos/maths/complex.hh>
+#include <eos/utils/exception.hh>
+
+#include <gsl/gsl_blas.h>
+#include <gsl/gsl_linalg.h>
+#include <gsl/gsl_matrix.h>
+
+#include <array>
+
+namespace eos
+{
+    // Abstract class implementing the algorithm of [M:1999A] to solve the Omnes integral equation
+    template <unsigned order_, unsigned nints_>
+    class OmnesFactor
+    {
+        protected:
+            // Vector containing integral borders
+            std::array<double, nints_> _intervals;
+
+            // Matrix holding the system of equations
+            gsl_matrix * _sys;
+
+            // Ingredients for SVD
+            gsl_matrix * _sys2;
+            gsl_matrix * _V;
+            gsl_vector * _S;
+            gsl_vector * _work;
+            gsl_vector * _x;
+            gsl_vector * _b;
+
+            // Vector containing Gauss-Legendre weights
+            std::array<double, order_> _weights;
+
+            // Vector containing zeros of Legendre Polynomials
+            std::array<double, order_> _zeros;
+
+            // Vector containing solution weights
+            std::array<double, nints_ * order_> _sol;
+
+            // Error of approximation
+            double _err = -1.;
+
+            // Did we already compute a solution?
+            bool _solved = false;
```

Comment:
Is this purely binary? Can the status change from ``_solved == true`` to ``_solved == false``? If not, let's solve this in the constructor. Being constructed then serves the same purpose :-)

Reply:
Good point, after solve_sys or set_weights is called it will be true no matter what.